app/test-utils/testFunctions/questionTests/errorTests.js

```diff
@@ -0,0 +1,154 @@
+    await pageSetup({ t });
+    const questionType = sectionManifest.questions[questionId].type;
+    let modifiedQuestionId = questionId;
+    // bulletpoint-list numbers the input fields so it is not just the question id e.g. listing-1
+    if (questionType === 'bulletpoint-list') modifiedQuestionId = `${questionId}-1`;
+    nock(apiLocalhost)
+      .put(sectionApiUrl)
+      .reply(400, {
+        [errorType]: [modifiedQuestionId],
+      });
+
+    const errorSummary = Selector('[data-test-id="error-summary"]');
+    const errorSummaryList = Selector('.nhsuk-error-summary__list');
+    const submitButton = Selector('[data-test-id="section-submit-button"]');
+
+    await t
+      .expect(errorSummary.exists).notOk()
+      .click(submitButton.find('button'))
+      .expect(errorSummary.exists).ok()
+      .expect(errorSummaryList.find('li:nth-child(1) a').count).eql(1)
+      .expect(errorSummaryList.find('li:nth-child(1) a').getAttribute('href')).eql(`#${modifiedQuestionId}`)
+      .click(errorSummaryList.find('li:nth-child(1) a'))
+      .expect(getLocation()).contains(`/solution/S100000-001/section/${sectionId}#${modifiedQuestionId}`);
+  });
+};
+
+const maxLengthErrorTest = ({
+  pageSetup,
+  sectionManifest,
+  questionId,
+  sectionApiUrl,
+  apiLocalhost,
+}) => {
+  test(`should show error summary and validation for ${questionId} question when it exceeds the maxLength`, async (t) => {
+    const questionType = sectionManifest.questions[questionId].type;
+    await pageSetup({ t });
+    nock(apiLocalhost)
+      .put(sectionApiUrl)
+      .reply(400, {
+        // bulletpoint-list numbers the input fields so it is not just the question id
+        maxLength: [questionType === 'bulletpoint-list' ? `${questionId}-1` : questionId],
+      });
+
+    const errorSummary = Selector('[data-test-id="error-summary"]');
+    const errorSummaryList = Selector('.nhsuk-error-summary__list');
+    const renderedQuestion = Selector(`[data-test-id="question-${questionId}"]`);
+    const submitButton = Selector('[data-test-id="section-submit-button"]');
+    const expectedErrorMessage = sectionManifest.questions[questionId].errorResponse.maxLength;
+
+    await t
+      .expect(errorSummary.exists).notOk()
+      .typeText(renderedQuestion, `${'x'.repeat(501)}`, { paste: true })
+      .click(submitButton.find('button'))
+      .expect(errorSummary.exists).ok()
+      .expect(errorSummaryList.find('li').count).eql(1)
+      .expect(errorSummaryList.find('li:nth-child(1)').innerText).eql(expectedErrorMessage)
+      .expect(errorSummaryList.find('li:nth-child(1) a').getAttribute('href')).eql(`#${questionType === 'bulletpoint-list' ? `${questionId}-1` : questionId}`)
+      .expect(renderedQuestion.find('.nhsuk-error-message').innerText).eql(`Error:\n${expectedErrorMessage}`);
+    if (questionType === 'textarea-field') {
+      await t
+        .expect(renderedQuestion.find('.nhsuk-textarea--error').exists).ok();
```

Comment:
this makes me a little sad that we have to do something different for a textarea. Is there something in our control that we can check the question is in error mode?

Reply:
🤷‍♀ that is the class that gives the red border for the text area

Reply:
🤔 in which case should we be checking the other questionTypes have the red border then?

Reply:
😄I'll have a look

Reply:
the only ones that have max length are textarea-field and text-field so I've added one for text-field